BranchStrategy (#6)
jabardigitalservice/development-guides: git_workflow.md

TBD.

## Branch Strategy
### Workflow
study case: master branch for production, development for staging.
origin/master -> local/master
origin/development -> local/development (created from master branch)
origin/feature or origin/bugfix -> lcoal/feature or local/bugfix (created from development branch)
* USE development as DEFAULT branch 
* DO NOT merge request / force push directly from local or origin feature/bugfix to origin master
* create merge request to master branch ONLY FROM origin development branch
* creeate merge request task to development branch ONLY

> * in case if hotfix needed, create branch hotfix/TaskName and merge it to master and development

Reviewer: JIka butuh hotfix, apakah base branch nya boleh langsung dari branch `master`? Atau base branchnya tetap harus dari branch `development`, seperti yang dijelaskan di poin 1?

Author: kalau pendapat saya pribadi misal hotfixnya case di production gpp narik hotfix dari master, point pentingnya di merge ke development dan master. #cmiiw
cc: @adrianpdm

Reviewer: Oke kang. Nanti boleh ditambahkan di poin di atas kang, misalnya jadi begini

```suggestion
* in case a production hotfix is needed, create branch `hotfix/TaskName` from branch `master`  and merge it to `master` and `development`.
```

### Branch Naming
suggestion:
1. Create branch depend on task given on trello / another task board, example:
    in trello   : #176 Migrasi Sesi 2
    branch name : MigrasiSesi2

>     * use PascalCase for branch name

Reviewer: Untuk branch naming, saya biasanya pakai `kebab-case`, karena tidak perlu pencet <kbd>Shift</kbd>, sehingga jumlah keystrokenya lebih sedikit. Selain itu, penggunaan hyphen `-` pada kebab case membuat setiap kata lebih mudah dibaca.

Tapi ini pendapat subjekif saya ya. Dan di luar sana, setau saya tidak ada standar baku untuk branch naming. Silakan kalau ada pendapat lain.

Author: terimakasih kang untuk itulah saya sediakan poin 1 2 3 supaya nanti kita bisa tau, so far kita bagaimana sih tentang penamaan branch ini kita menerapkan standar masing" seperti apa atau bahkan mungkin masih belum ada standar yang diterapkan, lalu kemudian saling diskusi untuk menyepakati satu bentuk (mungkin) yang dipakai sebagai standar kita bersama. Terimakasih untuk responsenya kang, ditunggu dari yang lain 😄 🙏
